File: intro/jee/jsf/examples/src/main/java/org/tsdes/intro/jee/jsf/examples/ex03/Comment.java
package org.tsdes.intro.jee.jsf.examples.ex03;

import javax.persistence.*;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Size;
import java.util.Date;

@Table(name = "Ex03_Comment")
@Entity
public class Comment {

    public static final int MIN_SIZE = 1;
    public static final int MAX_SIZE = 140;

    @Id
    @GeneratedValue
    private Long id;

    @NotNull
    @Size(min = MIN_SIZE, max = MAX_SIZE)
    private String text;

    @Temporal(TemporalType.TIMESTAMP)
    private Date date;


    public Comment(){}

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getText() {
        return text;
    }

    public void setText(String text) {
        this.text = text;
    }

    public Date getDate() {
        return date;
    }

    public void setDate(Date date) {
        this.date = date;
    }
}

File: advanced/rest/pagination/src/main/kotlin/org/tsdes/advanced/rest/pagination/entity/Comment.kt
package org.tsdes.advanced.rest.pagination.entity

import javax.persistence.Entity
import javax.persistence.GeneratedValue
import javax.persistence.Id
import javax.persistence.ManyToOne


@Entity
class Comment(

        @get:Id
        @get:GeneratedValue
        var id: Long? = null,

        var text: String?,

        @get:ManyToOne
        var news: News? = null
)